components/WithdrawLiquidity/WithdrawLiquidityController.tsx

```diff
@@ -1,26 +1,269 @@
+      if (accountInfo && accountInfo?.address) {
+        setdepPool(new ethers.Contract(address, DepPool.abi, provider.getSigner(accountInfo?.address)))
+      } else {
+        setdepPool(new ethers.Contract(address, DepPool.abi, provider))
+      }
+    } else {
+      console.log('Please connect wallet')
+    }
+
+    if (!accountInfo || !accountInfo.address) {
+      console.log('Wallet not connected.')
+      return
+    }
+
+    if (depPool === null) {
+      return
+    }
+
+    try {
+      let tx = await depPool.removeLiquidity(amt, ZEROMIN, ZEROMIN, accountInfo.address, {
+        gasLimit: 10000000,
+      })
+      return await tx.wait()
+    } catch (e) {
+      return e
+    }
+  }
+
+  async function approveWithdraw(depPool: Contract | null, depPoolAddr: string) {
+    if (!accountInfo || !accountInfo.address) {
+      console.log('Wallet not connected.')
+      return
+    }
+    if (depPool === null) {
+      return null
+    } else {
+      if (provider !== null) {
+        try {
+          let tx = await depPool.approve(depPoolAddr, constants.MaxUint256.toString())
+          return await tx.wait()
+        } catch (e) {
+          return e
+        }
+      } else {
+        console.log('Please connect wallet')
+      }
+    }
+  }
+
+  async function approve(fromToken: string, toAddr: string) {
+    if (!provider) {
+      console.log('provider or accountInfo not set')
+      return
+    }
+    if (!accountInfo || !accountInfo.address) {
+      console.log('Wallet not connected.')
+      return
+    }
+    if (depPool === null) {
+      return
+    }
+    var erc20 = new ethers.Contract(fromToken, IERC20.abi, provider.getSigner(accountInfo?.address))
+    let allowance = await erc20
+      .allowance(accountInfo.address, toAddr)
+      .then((res: string) => {
+        console.log('check allowance ', res.toString())
+        return res
+      })
+      .catch((err: Error) => {
+        console.error('checkAllowance', err)
+      })
+    if (parseFloat(allowance.toString()) <= 0) {
+      await erc20.approve(toAddr, constants.MaxUint256)
+    }
+  }
+
+  useEffect(() => {
+    async function fetchContract() {
+      if (!provider) {
+        console.log('Please connect wallet.')
+        return
+      }
+
+      if (provider) {
+        // Deposit poll contract address
+        let address = '0x3eFadc5E507bbdA54dDb4C290cc3058DA8163152'
+        if (accountInfo && accountInfo?.address) {
+          let _depPool = new ethers.Contract(address, DepPool.abi, provider.getSigner(accountInfo?.address))
+          setdepPool(_depPool)
+          const token0Addr = await _depPool.token0()
+          const token1Addr = await _depPool.token1()
+          const _token0 = new ethers.Contract(token0Addr, IERC20Metadata.abi, provider.getSigner(accountInfo?.address))
+          const _token1 = new ethers.Contract(token1Addr, IERC20Metadata.abi, provider.getSigner(accountInfo?.address))
+          const symbol0 = await _token0.symbol()
```

Comment:
might not be the best way to initialize all of these variables. Separation of concerns is key

Reply:
Fixed it